next.json

```diff
@@ -414,9 +438,84 @@
             }
         ]
     },
+    "depth stencil state": {
+        "category": "object"
+    },
+    "depth stencil state builder": {
+        "category": "object",
+        "methods": [
+            {
+                "name": "get result",
+                "returns": "depth stencil state"
+            },
+            {
+                "name": "set depth enabled",
+                "args": [
+                    {"name": "depth enabled", "type": "bool"}
+                ]
+            },
+            {
+                "name": "set depth compare function",
+                "args": [
+                    {"name": "depth compare function", "type": "compare function"}
+                ]
+            },
+            {
+                "name": "set depth write",
+                "args" : [
+                    {"name": "depth write mode", "type": "depth write mode"}
+                ]
+            },
+            {
+                "name": "set stencil enabled",
+                "args": [
+                    {"name": "stencil enabled", "type": "bool"}
+                ]
+            },
+            {
+                "name": "set stencil operation",
+                "args": [
+                    {"name": "face", "type": "face"},
+                    {"name": "stencil failure operation", "type": "stencil operation"},
+                    {"name": "depth failure operation", "type": "stencil operation"},
+                    {"name": "stencil pass operation", "type": "stencil operation"}
+                ]
+            },
+            {
+                "name": "set stencil compare function",
+                "args": [
+                    {"name": "face", "type": "face"},
+                    {"name": "stencil compare function", "type": "compare function"}
+                ]
+            },
+            {
+                "name": "set stencil mask",
+                "args": [
+                    {"name": "face", "type": "face"},
```

Comment:
Read and write both exist in D3D but they can't be set for each face separately, sorry my comment wasn't very clear.